backend/inspirehep/matcher/api.py

```diff
@@ -129,16 +142,40 @@ def match_reference(reference, previous_matched_recid=None):
     return reference
 
 
-def match_reference_control_number(reference):
+def match_references(reference):
+    """Match a reference using inspire-matcher.
+
+    Args:
+        reference (dict): the metadata of a reference.
+    Returns:
+        list: list of matched recids or None.
+    """
+    if reference.get("curated_relation"):
+        try:
+            return [get_recid_from_ref(reference["record"])]
+        except KeyError:
+            return None
+
+    configs = get_reference_matcher_configs(reference)
+
+    matches = []
+    for config in configs:
+        matched_recids = [
+            matched_record["_source"]["control_number"]
+            for matched_record in match(reference, config)
+        ]
+        matches.extend(matched_recids)
+    matches = dedupe_list(matches)[0:5]
```

Comment:
we could use a `set` instead of iterating again to remove dups and just get the first 5

Reply:
Yes, I was thinking about it but I thought that we want to keep the order of matches